packages/gitbook/src/components/Header/SpacesDropdownMenuItem.tsx

```diff
@@ -6,12 +6,19 @@ import { joinPath } from '@/lib/paths';
 import { useCurrentPagePath } from '../hooks';
 import { DropdownMenuItem } from './DropdownMenu';
 
-function useVariantSpaceHref(variantSpaceUrl: string) {
+function useVariantSpaceHref(variantSpaceUrl: string, currentSpacePath: string, active = false) {
     const currentPathname = useCurrentPagePath();
 
+    if (!active && currentPathname.startsWith(currentSpacePath)) {
```

Comment:
I don't really understand the bug and the logic here.

But I guess it should not match the case `currentPathname = 'hello-world'` and `currentSpacePath = 'hello'`? 

```suggestion
    if (!active && currentPathname.startsWith(currentSpacePath + '/')) {
```
```suggestion
    if (!active && currentPathname.startsWith(currentSpacePath)) {
```
```suggestion
    if (!active && currentPathname.startsWith(currentSpacePath)) {
```